onnx/src/ops/ml/mod.rs

```diff
@@ -288,26 +288,24 @@ impl InferenceRulesOp for TreeEnsembleClassifier {
         check_input_arity(&inputs, 1)?;
         check_output_arity(&outputs, 2)?;
 
-        s.equals(&inputs[0].rank, 2)?;
-        s.equals(&outputs[0].rank, 1)?;
         s.equals(&outputs[0].rank, 1)?;
+        s.equals(&outputs[0].rank, inputs[1].rank.bex() - 1)?;
+        s.equals(&outputs[1].rank, &inputs[1].rank)?;
+
+        s.given(&inputs[0].rank, move |s, rank| {
+            if rank == 1 && rank != 2 {
+                bail!("First input rank must be 1 or 2");
+            }
+            if rank == 2 {
+                s.equals(&inputs[0].shape[0], &outputs[0].shape[0])?;
+                s.equals(&inputs[0].shape[0], &outputs[1].shape[0])?;
+            }
+            // FIXME: do we somehow know Nfeatures ?
```

Comment:
There's `TreeEnsemble.check_n_features()`?

There's no exact equals bound, but there's a >= bound which can be inferred from the featureids used in all trees.